webapp/platform/design-system/src/components/primitives/tag/tag.stories.tsx

```diff
@@ -0,0 +1,423 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import type {Meta, StoryObj} from '@storybook/react';
+import React from 'react';
+
+import Tag from './tag';
+ import TagGroup from './tag_group';
+import {BetaTag, BotTag, GuestTag} from './tag_presets';
+
+import WithTooltip from '../with_tooltip';
+
+// Mock Compass Icon for demonstration
+const MockIcon = ({size = 16}: {size?: number}) => (
+    <svg width={size} height={size} viewBox='0 0 24 24' fill='currentColor'>
+        <path d='M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z'/>
+    </svg>
+);
+
+const meta: Meta<typeof Tag> = {
+    title: 'DesignSystem/Primitives/Tag',
+    component: Tag,
+    tags: ['autodocs'],
+    argTypes: {
+        text: {
+            control: 'text',
+            description: 'The text content of the tag',
+        },
+        variant: {
+            control: 'select',
+            options: ['default', 'info', 'success', 'warning', 'danger', 'dangerDim', 'primary', 'secondary'],
+            description: 'The visual variant of the tag',
+        },
+        size: {
+            control: 'select',
+            options: ['xs', 'sm', 'md', 'lg'],
+            description: 'The size of the tag',
+        },
+        uppercase: {
+            control: 'boolean',
+            description: 'Whether to display text in uppercase',
+        },
+        fullWidth: {
+            control: 'boolean',
+            description: 'Whether the tag should take full width',
+        },
+        onClick: {
+            action: 'clicked',
+            description: 'Click handler for interactive tags',
+        },
+    },
+};
+
+export default meta;
+type Story = StoryObj<typeof Tag>;
+
+// Basic Examples
+export const Default: Story = {
+    render: () => (
+        <div style={{padding: '20px', background: '#f0f0f0'}}>
+            <Tag text='Default Tag' variant='default' size='xs' />
+        </div>
```

Comment:
```suggestion
        <Tag text='Default Tag' variant='default' size='xs' />
```

Reply:
Do we need the container with padding and background? I assume we don't and can remove it for all cases below